dashboards-observability/public/components/application_analytics/components/create.tsx

```diff
@@ -126,6 +162,64 @@ export const CreateApp = (props: CreateAppProps) => {
     setIsFlyoutVisible(false);
   };
 
+  const closeIntegrationFlyout = () => {
+    setIsFlyoutVisibleIntegration(false);
+  };
+
+  const openIntegrationFlyout = () => {
+    setIsFlyoutVisibleIntegration(true);
+  };
+
+  let integrationFlyout;
+  if (isFlyoutVisibleIntegration) {
```

Comment:
This is evaluated on every render. The data inside integrationFlyout is a simple constant. You can create a function - something like renderFlyoutInfo - or some meaningful name and return the JSX if the flag is true, otherwise return null.